### PR TITLE
Return sample message for login error in api context

### PR DESCRIPTION
**Pull request by a1012112796**
As title, will return a sample message instead of full html page.

### CODE REVIEW

File: modules/context/auth.go [hunk 1]
```diff
@@ -32,20 +32,40 @@ func Toggle(options *ToggleOptions) macaron.Handler {
...
 
+		isAPIPath := auth.IsAPIPath(ctx.Req.URL.Path)
+
 		// Check prohibit login users.
 		if ctx.IsSigned {
 			if !ctx.User.IsActive && setting.Service.RegisterEmailConfirm {
 				ctx.Data["Title"] = ctx.Tr("auth.active_your_account")
+				if isAPIPath {
+					ctx.JSON(403, map[string]string{
+						"message": "You should active your account.",
```
Comment (silverwind) [inline modules/context/auth.go]:
```suggestion
						"message": "You should activate your account.",
```

File: modules/context/auth.go [hunk 2]
```diff
@@ -32,20 +32,40 @@ func Toggle(options *ToggleOptions) macaron.Handler {
...
 
+		isAPIPath := auth.IsAPIPath(ctx.Req.URL.Path)
+
 		// Check prohibit login users.
 		if ctx.IsSigned {
 			if !ctx.User.IsActive && setting.Service.RegisterEmailConfirm {
 				ctx.Data["Title"] = ctx.Tr("auth.active_your_account")
+				if isAPIPath {
+					ctx.JSON(403, map[string]string{
+						"message": "You should activate your account.",
```
Comment (lafriks) [inline modules/context/auth.go]:
```suggestion
						"message": "Your account is not active.",
```

File: modules/context/auth.go [hunk 3]
```diff
@@ -32,20 +32,40 @@ func Toggle(options *ToggleOptions) macaron.Handler {
...
+					return
+				}
 				ctx.HTML(200, "user/auth/prohibit_login")
 				return
 			}
 
 			if ctx.User.MustChangePassword {
+				if isAPIPath {
+					ctx.JSON(403, map[string]string{
+						"message": "Your must change your password, link: " + setting.AppURL + "/user/change_password",
```
Comment (lafriks) [inline modules/context/auth.go]:
```suggestion
						"message": "You must change password. You can do it using link " + setting.AppURL + "/user/change_password",
```

**Comment by zeripath** [comment]
hmm... AFAICS `context.Toggle` is ever called on an API path. Are you sure this ever runs?

File: modules/context/auth.go [hunk 4]
```diff
@@ -32,20 +32,40 @@ func Toggle(options *ToggleOptions) macaron.Handler {
...
 
+		isAPIPath := auth.IsAPIPath(ctx.Req.URL.Path)
+
 		// Check prohibit login users.
 		if ctx.IsSigned {
 			if !ctx.User.IsActive && setting.Service.RegisterEmailConfirm {
 				ctx.Data["Title"] = ctx.Tr("auth.active_your_account")
+				if isAPIPath {
+					ctx.JSON(403, map[string]string{
+						"message": "Your account is not active.",
```
Comment (zeripath) [inline modules/context/auth.go]:
```suggestion
						"message": "This account is not activated.",
```

File: modules/context/auth.go [hunk 5]
```diff
@@ -32,20 +32,40 @@ func Toggle(options *ToggleOptions) macaron.Handler {
...
+					return
+				}
 				ctx.HTML(200, "user/auth/activate")
 				return
 			} else if !ctx.User.IsActive || ctx.User.ProhibitLogin {
 				log.Info("Failed authentication attempt for %s from %s", ctx.User.Name, ctx.RemoteAddr())
 				ctx.Data["Title"] = ctx.Tr("auth.prohibit_login")
+				if isAPIPath {
+					ctx.JSON(403, map[string]string{
+						"message": "Your account is prohibited to sign in, please contact your site administrator.",
```
Comment (zeripath) [inline modules/context/auth.go]:
```suggestion
						"message": "This account is prohibited from signing in, please contact your site administrator.",
```

File: modules/context/auth.go [hunk 6]
```diff
@@ -32,20 +32,40 @@ func Toggle(options *ToggleOptions) macaron.Handler {
...
+					return
+				}
 				ctx.HTML(200, "user/auth/prohibit_login")
 				return
 			}
 
 			if ctx.User.MustChangePassword {
+				if isAPIPath {
+					ctx.JSON(403, map[string]string{
+						"message": "You must change password. You can do it using link " + setting.AppURL + "/user/change_password",
```
Comment (zeripath) [inline modules/context/auth.go]:
```suggestion
						"message": "You must change your password. Change it at: " + setting.AppURL + "/user/change_password",
```

**Review by zeripath — APPROVED** [review]
I think these suggestions read a bit better but otherwise LGTM.

**Review by lafriks — APPROVED** [review]

**Pull request merged**